sycl/source/detail/pi.cpp

```diff
@@ -378,9 +379,15 @@ static void initializePlugins(std::vector<plugin> &Plugins) {
     std::cerr << "SYCL_PI_TRACE[all]: "
               << "No Plugins Found." << std::endl;
 
-  PiPlugin PluginInformation{
-      _PI_H_VERSION_STRING, _PI_H_VERSION_STRING, nullptr, {}};
-  PluginInformation.PiFunctionTable = {};
+  std::shared_ptr<PiPlugin> PluginInformation;
+  {
+    PiPlugin PluginInformationInstance{
+        _PI_H_VERSION_STRING, _PI_H_VERSION_STRING, nullptr, {}};
+    PluginInformationInstance.PiFunctionTable = {};
+
+    PluginInformation.reset(new PiPlugin);
+    *PluginInformation = PluginInformationInstance;
+  }
```

Comment:
Why can't we just allocate `PiPlugin` and assign it to `PluginInformation` directly?
```suggestion
  std::shared_ptr<PiPlugin> PluginInformation{new PiPlugin{_PI_H_VERSION_STRING, _PI_H_VERSION_STRING, /*Targets=*/nullptr, /*FunctionPointers=*/{}};
```

Reply:
Changed using `make_shared`.